### PR TITLE
Updated package infrastructure

### PR DESCRIPTION
**Pull request by astrofrog**
As for other packages such as pytest-openfiles, this:

* Switches to setuptools_scm for version management
* Adds a tox configuration
* Removes AppVeyor
* Updates the Travis configuration (in fact this is a straight copy/paste from pytest-openfiles, we might be able to use the exact same file for all pytest plugings)

### CODE REVIEW

**Comment by pllim** [comment]
Thanks! Not sure why Appveyor tried to run. It shouldn't have given the `appveyor.yml` is no more.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+[testenv]
+changedir = .tmp/{envname}
+description = run tests
+deps =
+    pytest46: pytest==4.6.*
+    pytest50: pytest==5.0.*
+    pytest51: pytest==5.1.*
+    pytest52: pytest==5.2.*
+    pytest53: pytest==5.3.*
```
Comment (pllim) [inline tox.ini:18]: What about pytest 6.x ?

**Review by pllim — COMMENTED** [review]
p.s. Not sure how I feel about mixing up the OS like that in the matrix. I have a habit of doing the matrix in Linux and only test the other ones with latest stable, but that's a personal style.

**Comment by astrofrog** [comment]
> p.s. Not sure how I feel about mixing up the OS like that in the matrix. I have a habit of doing the matrix in Linux and only test the other ones with latest stable, but that's a personal style.

That's what I used to do but then I realized the matrix could be made smaller in this way and it really shouldn't matter. I think I was also worried that we introduce a bias that we never check non-linux with older versions of packages.

**Comment by pllim** [comment]
I am more worries that we'll end up dealing with failures that are not our fault but OS specific, especially for Windows. But I heard Macs are moving to ARM chips too (?), so who knows.

**Pull request merged**